nuxeo-features/nuxeo-user-center/nuxeo-user-profile/src/main/java/org/nuxeo/ecm/user/center/profile/rest/UserProfileEnricher.java

```diff
@@ -48,46 +62,69 @@
 @Setup(mode = SINGLETON, priority = REFERENCE)
 public class UserProfileEnricher extends AbstractJsonEnricher<NuxeoPrincipal> {
 
+    public static final String NAME = "userprofile";
+
     private static final FastDateFormat FORMATTER = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss'Z'");
 
-    public static final String NAME = "userprofile";
+    /**
+     * @since 9.3
+     */
+    public static final String COMPATIBILITY_CONFIGURATION_PARAM = "nuxeo.userprofile.enricher.compatibility";
+
+    @Inject
+    private SchemaManager schemaManager;
 
     public UserProfileEnricher() {
         super(NAME);
     }
 
     @Override
     public void write(JsonGenerator jg, NuxeoPrincipal nuxeoPrincipal) throws IOException {
-
-        UserProfileService ups = Framework.getLocalService(UserProfileService.class);
+        UserProfileService ups = Framework.getService(UserProfileService.class);
         try (RenderingContext.SessionWrapper sw = ctx.getSession(null)) {
             DocumentModel up = ups.getUserProfileDocument(nuxeoPrincipal.getName(), sw.getSession());
-
             jg.writeFieldName(NAME);
             if (up == null) {
                 jg.writeNull();
+                return;
+            }
+
+            jg.writeStartObject();
+            ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
```

Comment:
Not sure, but maybe it can be injected instead of being looked up? (Same for other services in this class.)